library/src/main/java/com/alamkanak/weekview/MonthLoader.kt

```diff
@@ -0,0 +1,30 @@
+package com.alamkanak.weekview
+
+import java.util.*
+
+class MonthLoader(var onMonthChangeListener: MonthChangeListener?) : WeekViewLoader {
+
+    override fun toWeekViewPeriodIndex(instance: Calendar): Double {
+        return (instance.get(Calendar.YEAR) * 12).toDouble() + instance.get(Calendar.MONTH).toDouble() + (instance.get(Calendar.DAY_OF_MONTH) - 1) / 30.0
+    }
+
+    override fun onLoad(periodIndex: Int): List<WeekViewEvent>? {
+        return onMonthChangeListener!!.onMonthChange(periodIndex / 12, periodIndex % 12 + 1)
```

Comment:
What about this ? :)

Reply:
I think you are both correct and incorrect. For some reason the library has code to check it that when you use it, you must set a listener:

        if (weekViewLoader == null && !isInEditMode)
            throw IllegalStateException("You must provide a MonthChangeListener")

and then when you set it:

    var monthChangeListener: MonthLoader.MonthChangeListener?
        get() = if (weekViewLoader is MonthLoader) (weekViewLoader as MonthLoader).onMonthChangeListener else null
        set(monthChangeListener) {
            this.weekViewLoader = MonthLoader(monthChangeListener)
        }

So you can't escape it from being non null.

But why this requirement exists at all?
How odd...
I think it's better to remove the requirement, and check for null...

For now I will let it stay how it is.